controllers/alerts/secret.go

```diff
@@ -0,0 +1,104 @@
+package alerts
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/jwt"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+const (
+	secretName = "hco-bearer-auth"
+)
+
+type secretReconciler struct {
+	theSecret *corev1.Secret
+}
+
+func newSecretReconciler(namespace string, owner metav1.OwnerReference) *secretReconciler {
+	return &secretReconciler{
+		theSecret: NewSecret(namespace, owner),
+	}
+}
+
+func (r *secretReconciler) Kind() string {
+	return "Secret"
+}
+
+func (r *secretReconciler) ResourceName() string {
+	return secretName
+}
+
+func (r *secretReconciler) GetFullResource() client.Object {
+	return r.theSecret.DeepCopy()
+}
+
+func (r *secretReconciler) EmptyObject() client.Object {
+	return &corev1.Secret{}
+}
+
+func (r *secretReconciler) UpdateExistingResource(ctx context.Context, cl client.Client, resource client.Object, logger logr.Logger) (client.Object, bool, error) {
+	found := resource.(*corev1.Secret)
+	modified := false
+
+	logger.Info("creating bearer token")
+	token, err := jwt.CreateToken()
+	if err != nil {
+		logger.Info("failed to create bearer token", "message", err)
+		token = ""
+	}
+
+	logger.Info("bearer token created", "token", token)
```

Comment:
Please avoid printing out the token. It may be sensitive

Reply:
ofc